crates/store/re_protos/proto/rerun/v0/catalog.proto

```diff
@@ -0,0 +1,249 @@
+syntax = "proto3";
+
+package rerun.redap.v0;
+
+import "rerun/v0/common.proto";
+
+// The RerunCatalog service is a common API set exposed by:
+// - The Rerun Catalog Server
+// - Any Rerun Storage Node
+//
+// Some Important terminology:
+// - Resource: A resource is any data that can be mapped to a single recording or table in the Rerun data model.
```

Comment:
We'll probably also need concepts similar to _sources_ and _sinks_ to describe the different format conversions/coercions that might happen, e.g sending an `.mcap` file to a `.rrd` sink forces a conversion on write.